### PR TITLE
Feature/157/create cookie policy page

### PR DESCRIPTION
**Pull request by Tolik170**
Created cookie policy page.

image

### CODE REVIEW

File: src/components/title-with-description/title-with-description.styles.js [hunk 1]
```diff
@@ -6,7 +6,22 @@ export const styles = {
...
+  textWithDot: {
+    my: '20px',
+    fontWeight: '600',
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+    '&::before': {
+      content: '""',
+      display: 'inline-block',
+      background: '#000',
```
Comment (tarasSanotskyi) [inline src/components/title-with-description/title-with-description.styles.js:20]: use variables from palette

**Review by tarasSanotskyi — CHANGES_REQUESTED** [review]

File: src/pages/cookie-policy/cookie-policy.styles.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+export const styles = {
+  container: {
+    color: '#263238'
+  },
+  sectionTitle: {
+    '&:first-of-type': {
+      background: '#ECEFF1',
```
Comment (tarasSanotskyi) [inline src/pages/cookie-policy/cookie-policy.styles.js:7]: same use colors from palette

**Review by tarasSanotskyi — CHANGES_REQUESTED** [review]

File: src/containers/layout/GuestLayout.js [hunk 3]
```diff
@@ -15,9 +16,9 @@ const GuestLayout = () => {
       <Route element={ <Example /> } name="home" path={ routes.about.route } />
       <Route element={ <AuthPolicy /> } path={ errors.authPolicy.route } />
       <Route element={ <BadRequest /> } path={ errors.badRequest.route } />
+      <Route element={ <CookiePolicy /> } name="FindMentor" path={ routes.privacyPolicy.route } />
       <Route element={ <NotFound /> } path={ errors.notFound.route } />
       <Route element={ <InternalServerError /> } path={ errors.internalServerError.route } />
-      <Route element={ <Navigate to={ errors.notFound.route }  /> } path='*' />
```
Comment (rostyslaw27) [inline src/containers/layout/GuestLayout.js:20]: return back this route

**Review by rostyslaw27 — CHANGES_REQUESTED** [review]

File: src/containers/layout/GuestLayout.js (hunk 3 above)
Comment (rostyslaw27) [reply src/containers/layout/GuestLayout.js:20]: image
fix code smells

**Review by tarasSanotskyi — CHANGES_REQUESTED** [review]

File: src/pages/cookie-policy/CookiePolicy.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+            title={ t(item.title) }
+            titleVariant={ item.id === 1 ? sectionTitleVariant : cookieItemsTitleVariant }
+          />
+        </Box>
+      </Box>
+    )
+  })
+
+  return (
+    <Box data-testid="sectionBox">
```
Comment (rostyslaw27) [inline src/pages/cookie-policy/CookiePolicy.js:35]: this box is probably redundant

**Review by rostyslaw27 — CHANGES_REQUESTED** [review]

File: tests/unit/pages/cookie-policy/cookiePolicy.spec.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+describe('CookiePolicy component test', () => {
+  beforeEach(() => {
+    render(<CookiePolicy />)
+  })
+  it('should have section box', () => {
+    const sectionBox = screen.getByTestId('sectionBox')
+
+    expect(sectionBox).toBeInTheDocument()
+  })
```
Comment (rostyslaw27) [inline tests/unit/pages/cookie-policy/cookiePolicy.spec.js:12]: add an empty lines between test blocks

**Review by rostyslaw27 — CHANGES_REQUESTED** [review]

File: src/pages/cookie-policy/cookie-policy.styles.js [hunk 6]
```diff
@@ -1,36 +1,35 @@
...
-    m: '20px 0 15px 0',
+  titleWithDot: {
+    mb: '15px',
     fontWeight: '600',
     display: 'flex',
     alignItems: 'center',
     '&::before': {
       content: '""',
-      background: '#607D8B',
+      background: palette.primary[500],
```
Comment (rostyslaw27) [inline src/pages/cookie-policy/cookie-policy.styles.js:32]: change to 'primary.500'

Comment (rostyslaw27) [reply src/pages/cookie-policy/cookie-policy.styles.js:32]: don't import palette

**Review by rostyslaw27 — CHANGES_REQUESTED** [review]

**Review by RestingState — APPROVED** [review]

File: src/pages/cookie-policy/cookie-policy.styles.js [hunk 7]
```diff
@@ -0,0 +1,39 @@
+import palette from '~/styles/app-theme/app.pallete'
```
Comment (tarasSanotskyi) [inline src/pages/cookie-policy/cookie-policy.styles.js:1]: you don`t need it

File: src/pages/cookie-policy/cookie-policy.styles.js [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    mb: '30px'
+  },
+  titleWithDot: {
+    mb: '15px',
+    fontWeight: '600',
+    display: 'flex',
+    alignItems: 'center',
+    '&::before': {
+      content: '""',
+      background: palette.primary[500],
```
Comment (tarasSanotskyi) [inline src/pages/cookie-policy/cookie-policy.styles.js:32]: 'primary.500'

File: src/pages/cookie-policy/cookie-policy.styles.js [hunk 9]
```diff
@@ -0,0 +1,39 @@
+import palette from '~/styles/app-theme/app.pallete'
+
+export const styles = {
+  container: {
+    color: palette.text.primary
+  },
+  sectionTitle: {
+    '&:first-of-type': {
+      background: palette.basic.grey,
```
Comment (tarasSanotskyi) [inline src/pages/cookie-policy/cookie-policy.styles.js:9]: 'basic.grey'

**Review by tarasSanotskyi — CHANGES_REQUESTED** [review]

File: src/containers/layout/GuestLayout.js [hunk 10]
```diff
@@ -15,6 +16,7 @@ const GuestLayout = () => {
       <Route element={ <Example /> } name="home" path={ routes.about.route } />
       <Route element={ <AuthPolicy /> } path={ errors.authPolicy.route } />
       <Route element={ <BadRequest /> } path={ errors.badRequest.route } />
+      <Route element={ <CookiePolicy /> } name="FindMentor" path={ routes.privacyPolicy.route } />
```
Comment (maarekua) [inline src/containers/layout/GuestLayout.js:19]: Should it have the FindMentor name?

File: src/pages/cookie-policy/CookiePolicy.js [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+import { useTranslation } from 'react-i18next'
+import { styles } from './cookie-policy.styles'
+import { cookieItemsData } from './cookie-policy.constants'
+
+const CookiePolicy = () => {
+  const { t } = useTranslation()
+
+  const itemsData = cookieItemsData.map((item) => {
+    return (
+      <Box data-testid="cookieItems" key={ item.id } sx={ styles.sectionTitle }>
```
Comment (maarekua) [inline src/pages/cookie-policy/CookiePolicy.js:12]: If the first element needs to have other CSS styles, you should use CSS for that.

File: src/pages/cookie-policy/cookie-policy.constants.js [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+    ...styles.wrapper
+  }
+}
+
+export const cookieItemsData = [
+  {
+    id: 1,
+    title: 'cookiePolicyPage.cookiePolicy.title',
+    description: 'cookiePolicyPage.cookiePolicy.description',
+    titleStyles: sectionTitleVariant,
```
Comment (maarekua) [inline src/pages/cookie-policy/cookie-policy.constants.js:30]: These are the cookie items, you should move style from here.

File: src/components/title-with-description/title-with-description.styles.js (hunk 1 above)
Comment (Tolik170) [reply src/components/title-with-description/title-with-description.styles.js]: fix

File: src/pages/cookie-policy/cookie-policy.styles.js (hunk 2 above)
Comment (Tolik170) [reply src/pages/cookie-policy/cookie-policy.styles.js]: fix

File: src/containers/layout/GuestLayout.js (hunk 3 above)
Comment (Tolik170) [reply src/containers/layout/GuestLayout.js:20]: fix

File: src/components/title-with-description/TitleWithDescription.js [hunk 13]
```diff
@@ -1,16 +1,30 @@
...
+      <Typography sx={ { typography: titleVariant, ...style.title } }>
         { title }
       </Typography>
 
-      <Typography sx={ { ...descriptionStyles } }>
-        { description }
-      </Typography>
+      <Box sx={ style.subtitleWrapper }>
+        <Typography sx={ textWithDot && { ...style.textWithDot, typography: textWithDotVariant } }>
+          { textWithDot }
```
[the remark this answers is not in the document]
Comment (Tolik170) [reply src/components/title-with-description/TitleWithDescription.js]: fix

File: src/pages/cookie-policy/CookiePolicy.js (hunk 4 above)
Comment (Tolik170) [reply src/pages/cookie-policy/CookiePolicy.js]: fix

File: tests/unit/pages/cookie-policy/cookiePolicy.spec.js (hunk 5 above)
Comment (Tolik170) [reply tests/unit/pages/cookie-policy/cookiePolicy.spec.js:13]: fix

File: src/pages/cookie-policy/cookie-policy.styles.js (hunk 6 above)
Comment (Tolik170) [reply src/pages/cookie-policy/cookie-policy.styles.js]: fix

File: src/pages/cookie-policy/cookie-policy.styles.js (hunk 7 above)
Comment (Tolik170) [reply src/pages/cookie-policy/cookie-policy.styles.js]: fix

File: src/pages/cookie-policy/cookie-policy.styles.js (hunk 8 above)
Comment (Tolik170) [reply src/pages/cookie-policy/cookie-policy.styles.js]: fix

File: src/pages/cookie-policy/cookie-policy.styles.js (hunk 9 above)
Comment (Tolik170) [reply src/pages/cookie-policy/cookie-policy.styles.js]: fix

**Review by maarekua — CHANGES_REQUESTED** [review]

File: src/components/title-with-description/TitleWithDescription.js [hunk 14]
```diff
@@ -1,10 +1,10 @@
 import { Box, Typography } from '@mui/material'
 import { styles } from '~/components/title-with-description/title-with-description.styles'
 
-const TitleWithDescription = ({ title, titleStyles, description, descriptionStyles, componentStyles }) => {
+const TitleWithDescription = ({ title, titleStyles, description, descriptionStyles, componentStyles = styles }) => {
   return (
-    <Box sx={ { ...styles.wrapper, ...componentStyles } }>
-      <Typography sx={ { ...styles.title, ...titleStyles } }>
+    <Box sx={ { ...styles.wrapper, ...componentStyles.wrapper } }>
```
Comment (LubomyrKashuba) [inline src/components/title-with-description/TitleWithDescription.js:6]: seems like you set the same styles to this box

File: src/pages/cookie-policy/cookie-policy.styles.js [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+    m: '0 auto 40px',
+    maxWidth: '744px'
+  },
+  wrapper: {
+    textAlign: 'start'
+  },
+  title: {
+    mb: '30px'
+  },
+  titleWithDot: {
```
Comment (LubomyrKashuba) [inline src/pages/cookie-policy/cookie-policy.styles.js:23]: maybe will be better if you use <li> tag for that

Comment (Tolik170) [reply src/pages/cookie-policy/cookie-policy.styles.js:23]: I think it's better to just pass the text for the title in titleWithDescription component and create a dot by css, than use extra tags and maybe change the component.

File: src/components/title-with-description/TitleWithDescription.js [hunk 16]
```diff
@@ -1,10 +1,10 @@
...
+      <Typography sx={ { ...styles.title, ...titleStyles, ...componentStyles.title } }>
```
Comment (rostyslaw27) [inline src/components/title-with-description/TitleWithDescription.js:7]: pass title styles to titleStyles prop

File: src/containers/guest-home-page/forgot-password/ForgotPassword.js [hunk 17]
```diff
@@ -27,7 +27,7 @@ const ForgotPassword = () => {
   return (
     <Box sx={ style.root }>
       <TitleWithDescription
-        componentStyles={ style.wrapper }
+        componentStyles={ style }
```
Comment (rostyslaw27) [inline src/containers/guest-home-page/forgot-password/ForgotPassword.js:30]: discard this change

File: src/containers/guest-home-page/reset-password-success/ResetPasswordSuccess.js [hunk 18]
```diff
@@ -42,7 +42,7 @@ const ResetPasswordSuccess = ({ email }) => {
       />
 
       <TitleWithDescription
-        componentStyles={ style.wrapper }
+        componentStyles={ style }
```
Comment (rostyslaw27) [inline src/containers/guest-home-page/reset-password-success/ResetPasswordSuccess.js:45]: discard this change, we don't have to pass all styles, and get only needed

File: src/pages/cookie-policy/CookiePolicy.js [hunk 19]
```diff
@@ -0,0 +1,33 @@
+import TitleWithDescription from '~/components/title-with-description/TitleWithDescription'
+import { Box, Container } from '@mui/material'
+import { useTranslation } from 'react-i18next'
```
Comment (rostyslaw27) [inline src/pages/cookie-policy/CookiePolicy.js:3]: reorganise imports:
-libs
-components
-styles
and add empty spaces between them

File: src/pages/cookie-policy/cookie-policy.constants.js [hunk 20]
```diff
@@ -0,0 +1,113 @@
...
+}
+const titleWithDotVariant = {
+  typography: { xlg: 'subtitle1' }
+}
+const descriptionVariant = {
+  typography: { md: 'body1', xs: 'body2' }
+}
+
+const titleWithDotStyles = {
+  title: {
```
Comment (rostyslaw27) [inline src/pages/cookie-policy/cookie-policy.constants.js:17]: maybe you can do
const { titleWithDots, wrapper } = styles

const titleWithDotStyles= {
   titleWithDots,
   wrapper
}

**Review by rostyslaw27 — CHANGES_REQUESTED** [review]

File: src/pages/cookie-policy/CookiePolicy.js (hunk 19 above)
Comment (Tolik170) [reply src/pages/cookie-policy/CookiePolicy.js:2]: fixed

File: src/containers/layout/GuestLayout.js (hunk 10 above)
Comment (Tolik170) [reply src/containers/layout/GuestLayout.js]: fixed

File: src/pages/cookie-policy/CookiePolicy.js (hunk 11 above)
Comment (Tolik170) [reply src/pages/cookie-policy/CookiePolicy.js]: fixed

File: src/pages/cookie-policy/cookie-policy.constants.js (hunk 12 above)
Comment (Tolik170) [reply src/pages/cookie-policy/cookie-policy.constants.js]: fixed

File: src/components/title-with-description/TitleWithDescription.js (hunk 14 above)
Comment (Tolik170) [reply src/components/title-with-description/TitleWithDescription.js]: fixed

File: src/components/title-with-description/TitleWithDescription.js (hunk 16 above)
Comment (Tolik170) [reply src/components/title-with-description/TitleWithDescription.js]: fixed

File: src/containers/guest-home-page/forgot-password/ForgotPassword.js (hunk 17 above)
Comment (Tolik170) [reply src/containers/guest-home-page/forgot-password/ForgotPassword.js]: fixed

File: src/containers/guest-home-page/reset-password-success/ResetPasswordSuccess.js (hunk 18 above)
Comment (Tolik170) [reply src/containers/guest-home-page/reset-password-success/ResetPasswordSuccess.js]: fixed

File: src/pages/cookie-policy/cookie-policy.constants.js (hunk 20 above)
Comment (Tolik170) [reply src/pages/cookie-policy/cookie-policy.constants.js]: fixed

File: src/components/title-with-description/TitleWithDescription.js [hunk 21]
```diff
@@ -1,13 +1,17 @@
...
 
-const TitleWithDescription = ({ title, titleStyles, description, descriptionStyles, componentStyles }) => {
+const TitleWithDescription = ({ title, titleStyles, subtitle, subtitleStyles, description, descriptionStyles, componentStyles }) => {
   return (
     <Box sx={ { ...styles.wrapper, ...componentStyles } }>
       <Typography sx={ { ...styles.title, ...titleStyles } }>
         { title }
       </Typography>
 
+      <Typography sx={ subtitle && { ...styles.subtitle, ...subtitleStyles } }>
```
Comment (LubomyrKashuba) [inline src/components/title-with-description/TitleWithDescription.js:11]: maybe you should create new component that will fit to your structure

Comment (maarekua) [reply src/components/title-with-description/TitleWithDescription.js:11]: I remember that I have told not to reuse this component and add any subtitles to it.

Comment (maarekua) [inline src/components/title-with-description/TitleWithDescription.js:11]: I am still waiting for changes.

**Review by maarekua — CHANGES_REQUESTED** [review]

File: src/components/title-with-description/TitleWithDescription.js (hunk 21 above)
Comment (Tolik170) [reply src/components/title-with-description/TitleWithDescription.js]: done.

Comment (Tolik170) [reply src/components/title-with-description/TitleWithDescription.js]: done.

**Review by maarekua — APPROVED** [review]

**Review by tarasSanotskyi — APPROVED** [review]

File: src/pages/cookie-policy/CookiePolicy.js [hunk 22]
```diff
@@ -0,0 +1,45 @@
...
+  const { t } = useTranslation()
+
+  const cookieItems = cookieItemsData.map((item, index) => {
+    return (
+      <Box key={ index } sx={ styles.wrapper }>
+        <Typography sx={ styles.titleStyles }>
+          { t(item.title) }
+        </Typography>
+
+        <Typography sx={ item.subtitle && styles.subtitleStyles }>
```
Comment (rostyslaw27) [inline src/pages/cookie-policy/CookiePolicy.js:18]: just styles.subtitle

File: src/pages/cookie-policy/cookie-policy.styles.js [hunk 23]
```diff
@@ -0,0 +1,47 @@
+export const styles = {
+  container: {
+    color: 'primary.900'
+  },
+  wrapper: {
+    m: '0 auto',
+    textAlign: 'start',
```
Comment (rostyslaw27) [inline src/pages/cookie-policy/cookie-policy.styles.js:7]: text align start is by default

File: src/pages/cookie-policy/cookie-policy.styles.js [hunk 24]
```diff
@@ -0,0 +1,47 @@
...
+  titleStyles: {
+    typography: { md: 'h5', xs: 'h5' },
+    mb: '25px',
+  },
+  subtitleStyles: {
+    typography: { xlg: 'subtitle1' },
+    fontWeight: 600,
+    my: '15px',
+    display: 'flex',
+    justifyContent: 'start',
```
Comment (rostyslaw27) [inline src/pages/cookie-policy/cookie-policy.styles.js:33]: justifyContent start is by default

**Review by rostyslaw27 — CHANGES_REQUESTED** [review]

**Review by rostyslaw27 — APPROVED** [review]

File: src/pages/cookie-policy/cookie-policy.styles.js (hunk 24 above)
Comment (Tolik170) [reply src/pages/cookie-policy/cookie-policy.styles.js]: fixed

File: src/pages/cookie-policy/CookiePolicy.js (hunk 22 above)
Comment (Tolik170) [reply src/pages/cookie-policy/CookiePolicy.js]: fixed

File: src/pages/cookie-policy/cookie-policy.styles.js (hunk 23 above)
Comment (Tolik170) [reply src/pages/cookie-policy/cookie-policy.styles.js]: fixed

**Pull request merged**